scripts/variantstore/wdl/GvsAssignIds.wdl

```diff
@@ -0,0 +1,168 @@
+version 1.0
+
+workflow GvsAssignIds {
+
+  input {
+    Array[String] external_sample_names
+    String project_id
+    String dataset_name
+    String sample_info_table = "sample_info"
+    String sample_info_schema = "sample_name:STRING,sample_id:INTEGER"
+    String? service_account_json
+    Int? preemptible_tries
+    File? gatk_override
+    String? docker
+  }
+
+  String docker_final = select_first([docker, "us.gcr.io/broad-gatk/gatk:4.1.7.0"])
+
+
+  call CreateTables as CreateSampleInfoTables {
+  	input:
+      project_id = project_id,
+      dataset_name = dataset_name,
+      datatype = "sample_info",
+      schema = sample_info_schema,
+      service_account_json = service_account_json,
+      preemptible_tries = preemptible_tries,
+      docker = docker_final
+  }
+
+  call AssignIds {
+    input:
+      sample_names = external_sample_names,
+      project_id = project_id,
+      dataset_name = dataset_name,
+      sample_info_table = sample_info_table,
+      table_creation_done = CreateSampleInfoTables.done,
+      gatk_override = gatk_override,
+      service_account_json = service_account_json,
+      docker = docker_final,
+  }
+  
+  output {
+    Boolean gvs_ids_created = true
+  }
+}
+
+task AssignIds {
+  input {
+    Array[String] sample_names
+    String project_id
+    String dataset_name
+    String sample_info_table
+    String? service_account_json
+    String table_creation_done
+    # runtime
+    File? gatk_override
+    String docker
+
+  }
+
+  String has_service_account_file = if (defined(service_account_json)) then 'true' else 'false'
+
+  meta {
+    description: "Assigns Ids to samples"
+    volatile: true
+  }
+
+  command <<<
+      set -e
+
+      export GATK_LOCAL_JAR=~{default="/root/gatk.jar" gatk_override}
+
+      if [ ~{has_service_account_file} = 'true' ]; then
+        gsutil cp ~{service_account_json} local.service_account.json
+        gcloud auth activate-service-account --key-file=local.service_account.json
+      fi
+
+      echo "project_id = ~{project_id}" > ~/.bigqueryrc
+
+      # create the lock table
+      bq --project_id=~{project_id} mk ~{dataset_name}.metadata_lock
```

Comment:
just checking, this script exits with an error of the metadata_lock table already exists?

Reply:
also -- maybe we should name this differently now that this isn't going into the "metadata" table?  sample_info_lock or sample_id_assignment_lock?

Reply:
also also… I'm surprised that this doesn't have to specify the schema when the table is created?

Reply:
will explicitly check whether the table exists and exit with error if it does. 
renaming to sample_id_assignment_lock
adding schema when creating